### ISSUE TITLE
[source-quickbooks] 'str' object has no attribute 'eval'  ERROR

### ISSUE THREAD

**Issue by solidpro-devops** [airbytehq/airbyte#47305]
### Connector Name

Quickbooks

### Connector Version

3.0.25

### What step the error happened?

Configuring a new connector

### Relevant information

The connector was working for me last month, but it was disabled due to a separate issue. After noticing a new version of the connector was available, I re-enabled the sync, but it failed again with the same error: 'str' object has no attribute 'eval'.

I deleted the existing connection/source and set up a new source, hoping that would resolve the issue. However, the same error occurred when testing the connection

### Relevant log output

```shell
2024-10-23 15:43:20 [46mplatform[0m > 
2024-10-23 15:43:20 [46mplatform[0m > Connector exited, processing output
2024-10-23 15:43:20 [46mplatform[0m > ----- START CHECK -----
2024-10-23 15:43:20 [46mplatform[0m > 
2024-10-23 15:43:20 [46mplatform[0m > Output file jobOutput.json found
2024-10-23 15:43:20 [46mplatform[0m > Connector exited with exit code 0
2024-10-23 15:43:20 [46mplatform[0m > Reading messages from protocol version 0.2.0
2024-10-23 15:43:20 [46mplatform[0m > MetricClient has not been initialized. Must call MetricClientFactory.CreateMetricClient before using MetricClient. Using a dummy client for now. Ignore this if Airbyte is configured to not publish any metrics.
2024-10-23 15:43:20 [46mplatform[0m > Encountered an error trying to connect to stream accounts. Error: 
 Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/checks/check_stream.py", line 42, in check_connection
    stream_is_available, reason = availability_strategy.check_availability(stream, logger, source)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/streams/http/availability_strategy.py", line 50, in check_availability
    get_first_record_for_slice(stream, stream_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/streams/utils/stream_helper.py", line 40, in get_first_record_for_slice
    return next(records_for_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py", line 126, in read_records
    yield from self.retriever.read_records(self.get_json_schema(), stream_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 339, in read_records
    self.cursor.close_slice(_slice, most_recent_record_from_slice)
  File "/airbyte/integration_code/source_quickbooks/components.py", line 72, in close_slice
    most_recent_record=LastRecordDictProxy(most_recent_record, {self.cursor_field.eval(self.config): "MetaData/LastUpdatedTime"}),
AttributeError: 'str' object has no attribute 'eval'

2024-10-23 15:43:20 [46mplatform[0m > Check failed
2024-10-23 15:43:20 [46mplatform[0m > Checking for optional control message...
2024-10-23 15:43:20 [46mplatform[0m > Optional control message not found. Skipping...
2024-10-23 15:43:20 [46mplatform[0m > Writing output of cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_19ef2754-888c-44ef-9bc8-b5ef0913655c_0_check to the doc store
2024-10-23 15:43:20 [46mplatform[0m > Marking workload cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_19ef2754-888c-44ef-9bc8-b5ef0913655c_0_check as successful
2024-10-23 15:43:21 [46mplatform[0m > 
2024-10-23 15:43:21 [46mplatform[0m > Deliberately exiting process with code 0.
2024-10-23 15:43:21 [46mplatform[0m > ----- END CHECK -----
2024-10-23 15:43:21 [46mplatform[0m >
```

### Contribute

- [ ] Yes, I want to contribute

**Comment by cmhnk** [comment]
I'm experiencing this same issue, and also decided to delete the original source and connection and configure a new source/connection. Confirming that re-configuration did not solve the issue.

FWIW I think it may be related a series of releases & reversions on October 14 -- this is the date that failures began for us. 

Original issue: [[source-quickbooks] Trying to set-up quickbooks #44367](https://github.com/airbytehq/airbyte/issues/44367)

PR that fixed the original issue: [🐛 Source Quickbooks: Replace Custom Components with Airbyte CDK #44560](https://github.com/airbytehq/airbyte/pull/44560). 

- This was committed 10/14 with docker image tag `3.0.26-rc.1`. 

PR that reverts the previous PR: [Revert "🐛 Source Quickbooks: Replace Custom Components with Airbyte CDK" #46882](https://github.com/airbytehq/airbyte/pull/46882). 

- This reverted the docker image tag from `3.0.26-rc.1` to `3.0.25`. 

PR that reverts the reversion PR: (Revert "Revert "🐛 Source Quickbooks: Replace Custom Components with Airbyte CDK"" #46885)[https://github.com/airbytehq/airbyte/pull/46885/files]

- This reverted the docker image tag from `3.0.25` to `3.0.26-rc.1`. 

Would a new release (e.g. `3.0.26`) ensure that all Airbyte Cloud accounts using this connector have the latest code?

**Comment by marcosmarxm** [comment]
@natikgadzhi and @pabloescoder can you take a look? This problem should have been solved in #44367, but it seems it hasn't been addressed yet.

**Comment by pabloescoder** [comment]
@marcosmarxm Sure, I'll look into what's wrong.

**Comment by mtdewulf** [comment]
Any update on this?  Quickbooks is dead to us until this gets resolved.

Or does anyone have a workaround?

**Comment by natikgadzhi** [comment]
@mtdewulf this is on my radar, but I can't yet get enough time to help unblock @pabloescoder ;-(

**Comment by niteshsharoff** [comment]
+1 on the issue. QB throws this error and doesn't extract/load. :(

**Comment by solidpro-devops** [comment]
Quick bump on this!

**Comment by mtdewulf** [comment]
We're just running this with the open source version for now.  Very frustrated that the cloud version doesn't work.

**Comment by g-hartmann-kanguro** [comment]
@mtdewulf would you be willing to share how you got the OS version working?

**Comment by mtdewulf** [comment]
@g-hartmann-kanguro, I had to downgrade the connector to v3.0.3 and it worked.

**Comment by pabloescoder** [comment]
@solidpro-devops @mtdewulf @niteshsharoff @g-hartmann-kanguro Just wanted to confirm, which version are you using? 
Are you getting this error even in 3.0.26?

**Comment by mtdewulf** [comment]
This does seem to be working with 3.0.26 now.

**Comment by pabloescoder** [comment]
Great! 3.0.26 has the fix, so it should work for everyone.

**Comment by niteshsharoff** [comment]
Confirmed for me too, 3.0.26 synced correctly

**Comment by marcosmarxm** [comment]
@pabloescoder feel free to close the issue as it looks the problem is solved.

**Comment by pabloescoder** [comment]
@marcosmarxm Sure, sounds good. I'll close it.
